controllers/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -155,6 +178,52 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	return reconcile.Result{}, nil
 }
 
+func (r *Reconciler) isUserBanned(ctx context.Context, userSignup *toolchainv1alpha1.UserSignup) (bool, error) {
+	banned := false
+	// Lookup the user email
+	if userSignup.Spec.IdentityClaims.Email != "" {
+
+		// Lookup the email hash label
+		if emailHashLbl, exists := userSignup.Labels[toolchainv1alpha1.UserSignupUserEmailHashLabelKey]; exists {
+
+			labels := map[string]string{toolchainv1alpha1.BannedUserEmailHashLabelKey: emailHashLbl}
+			opts := runtimeclient.MatchingLabels(labels)
+			bannedUserList := &toolchainv1alpha1.BannedUserList{}
+
+			// Query BannedUser for resources that match the same email hash
+			if err := r.Client.List(ctx, bannedUserList, opts); err != nil {
+				return false, err
+			}
+
+			// One last check to confirm that the e-mail addresses match also (in case of the infinitesimal chance of a hash collision)
+			for _, bannedUser := range bannedUserList.Items {
+				if bannedUser.Spec.Email == userSignup.Spec.IdentityClaims.Email {
+					banned = true
+					break
+				}
+			}
+
+			hashIsValid := validateEmailHash(userSignup.Spec.IdentityClaims.Email, emailHashLbl)
+			if !hashIsValid {
+				err := fmt.Errorf("hash is invalid")
+				return banned, err
+			}
+		} else {
+			// If there isn't an email-hash label, then the state is invalid
+			err := fmt.Errorf("missing label at usersignup")
```

Comment:
Let's add the usersignup name to the error message.

Reply:
Fixed